src/mtype.d

```diff
@@ -8952,14 +8934,6 @@ public:
             e = e.semantic(sc);
             return e;
         }
-        if (d.parent && d.toParent().isModule())
```

Comment:
If `d` is VarDeclaration, `d.isDataseg()` would return true, and it's already handled in previous lines.
If `d` is FuncDeclaration, we have to consider it would have overloads. For example:

```d
module x;
void foo() {}

class C
{
    alias bar = .foo;
    void bar(int) {}
}

void main()
{
    auto c = new C();
    c.bar(1);
}
```

By removing this, such the dificult case handlings will be gathered into `DotVarExp.semantic`.